### ISSUE TITLE
Proposal for Greed Island 2.0

### ISSUE THREAD

**Issue by Kile** [Kile/Killua#631]
# Proposal for the largest update to the Greed Island system in a while

I am currently rewatching hunter x hunter and got a bunch of new ideas on how to improve the current system and make it more interactive and fun. The core idea is to be able to "enter" and "leave" the game.

## Entering & leaving the game

Entering the game should now require a Greed Island game copy which can either be bought for a fixed price or maybe in a more interactive auction system. This among other changes detailed below will require a rethink of the inventory design. Once obtained, people can enter the game at any time provided a limit of players is not reached.

This allows for a number of changes:
- Since entering the game is now an extra step, so is traveling. Travel spell cards like "Accompany" will be implemented to travel between cities in the game
- The cities will follow the cities from the anime. To buy spell cards, you now need to go to Masadora etc. 
- You can travel between cities by foot using a new command `/travel` which will take a while to complete
- This speed can be improved by training, which will be elaborated on further below
- The "leave" spell and port will be implemented as ways to leave the game
- Leaving the game erases free slots and leaving it for longer than 14 days will erase restricted slot cards
- After a set amount of time, Killua should kick inactive players from the game to prevent lots of inactive users clogging up the card and game limits.

## Training

I have wanted a reason for `gain` to be useful for a while. I think the best way is to require a certain set of items (gained) for training or similar interactions in a way that makes sense. For example, to train strength, you'd need 10 "gained" rocks to shatter, similarly to how Gon and Killua use rocks to train in the anime.

This would also allow support for some more special interactions like Paladin's Necklace or Risky Dice to finally be implemented whereas before it seemed strange to just implement those two to be "gainable". 

Training would allow you to increase stats such as speed, important for `/travel`, or strength which could be used for `/hunt` or unique challenges to gain cards. Speaking of unique challenges...

## Changes to obtaining rare cards

Being only able to obtain S and SS cards from loot boxes was never ideal. Adding the "physical" aspect to the game, will allow for some more unique challenges like fighting the pirate crew and then Razor and even custom challenges by traveling to certain places. As a result, loot boxes will need to be reworked and currently obtained S or SS cards (not spells) will be refunded to players who had already obtained them, to make the game not just about buying cards but more of a challenge.

This system will definitely not include all cards right from the getgo as to not rush challenges and make them repetitive and boring. That also gives me some more time to work on a more fun event when someone collects all 99 cards :)
(Which by the way is another thing I have been thinking about. The current triggered event is not very exciting, but this would open up the opportunity to be more similar to the anime and a more exciting goal for players to work towards)

## Utilising Discord Activities

For a while now I have been closely eying Discord Activities which are proper games within Discord and thought about ways to integrate them into Killua. I am not very good at traditional game dev but if I find someone to help me this would be an insane cool thing to add to make the challenges a lot more interactive. If you have some experience, please reach out!

###  Lootbox shop
With loot boxes being reworked, the shop probably also will be. Possibly they will only be dropped and not be able to be bought now, making them more special with some other potential items unrelated to greed island (except Jenny)

### Card shop
The card shop will include a wider selection of cards and be moved to Masadora, only being accessible while being in game in Masadora.

### ToDo shop
ToDo lists are a very separate thing from the Greed Island and Jenny system and are also not used a lot. Instead of reworking the shop, it may be better to fully remove it and allow all features for todo lists you could buy for free.

### TLDR
The proposed changes are allowing people to enter and leave the game island, opening up possibilities for a bunch of new spell cards to be implemented as well as more elaborate challenges to obtain rare cards, progress in abilities and skill and a more active player base!
